gamemodes/terrortown/gamemode/server/sv_shop.lua

```diff
@@ -66,150 +57,49 @@ function GM:TTT2OrderedEquipment(ply, cls, isItem, credits, ignoreCost)
 
 end
 
-local function IsPartOfShop(ply, cls)
-	if not GetGlobalBool("ttt2_random_shops") or not RANDOMSHOP[ply] or #RANDOMSHOP[ply] == 0 then
-		return true
-	end
-
-	for i = 1, #RANDOMSHOP[ply] do
-		if RANDOMSHOP[ply][i].id ~= cls then continue end
-
-		return true
-	end
-
-	print(ply, "tried to buy a prohibited item/weapon!")
-
-	return false
-end
-
 -- Equipment buying
-local function OrderEquipment(ply, cls)
-	if not IsValid(ply) or not cls or not ply:IsActive() then return end
-
-	-- if we have a pending order because we are in a confined space, don't
-	-- start a new one
-	if HasPendingOrder(ply) then
-		LANG.Msg(ply, "buy_pending", nil, MSG_MSTACK_ROLE)
 
+local function HandleErrorMessage(ply, equipmentId, statusCode)
+	if statusCode == shop.statusCode.SUCCESS then
 		return
+	elseif statusCode == shop.statusCode.PENDINGORDER then
+		LANG.Msg(ply, "buy_pending", nil, MSG_MSTACK_ROLE)
+	elseif statusCode == shop.statusCode.NOTEXISTING then
+		print(ply, " tried to buy equip that doesn't exist: ", equipmentId)
+	elseif statusCode == shop.statusCode.NOTENOUGHCREDITS then
+		print(ply, " tried to buy item/weapon, but didn't have enough credits.")
+	elseif statusCode == shop.statusCode.INVALIDID then
+		ErrorNoHalt("[TTT2][ERROR] No ID was requested by:", ply)
+	elseif statusCode == shop.statusCode.NOTBUYABLE then
+		LANG.Msg(ply, "This equipment cannot be bought.", nil, MSG_MSTACK_ROLE)
+	elseif statusCode == shop.statusCode.NOTENOUGHPLAYERS then
+		LANG.Msg(ply, "Minimum amount of active players needed.", nil, MSG_MSTACK_ROLE)
+	elseif statusCode == shop.statusCode.LIMITEDBOUGHT then
+		LANG.Msg(ply, "This equipment is limited and is already bought.", nil, MSG_MSTACK_ROLE)
+	elseif statusCode == shop.statusCode.NOTBUYABLEFORROLE then
+		LANG.Msg(ply, "Your role can't buy this equipment.", nil, MSG_MSTACK_ROLE)
```

Comment:
I think we should localize these, at least these that are in `LANG.Msg` that already supports localization. I think they also should all be moved to the `MSTACK`, there is no reason to have two of them in chat and `LANG.Msg` also supports the parameter translation needed for `equipmentID`

Reply:
I thought that someone would say these. IMHO we shouldnt do this in this PR. I try to keep functionality exactly the same. Maybe fix an error if I see one. But generally I try just sort everything and we can still modernize it later. Makes thes PRs faster to review.